### ISSUE TITLE
Datetime widget incorrectly parses dates without times

### ISSUE THREAD

**Issue by nholden** [netlify/netlify-cms#3640]
I have a field that I would like to contain just a date (with no specific time). I saw that the [`date` widget](https://www.netlifycms.org/docs/widgets/date) is deprecated, so I reached for the [`datetime` widget](https://www.netlifycms.org/docs/widgets/datetime/). When I configure the `datetime` widget with sensible options for a date-only field, dates are stored properly in the saved markdown, but when I load those dates in the UI, I see the date before.

I think this is happening because [the `DateTime` component](https://github.com/netlify/netlify-cms/blob/b151bdfd7311472eaa1861e9fc17546a3658a083/packages/netlify-cms-widget-datetime/src/DateTimeControl.js#L120-L128) from the `react-datetime` library is using local timezones. It loads the date as the start of day UTC and then converts to the local timezone, which is going to be the previous day in any timezone with a negative UTC offset, including all of the Americas.

Is there a supported way to have a date-only field in Netlify CMS, or is this a bug?

I see that the `react-datetime` API has a [`utc` option](https://github.com/YouCanBookMe/react-datetime#api), which may be helpful here.

## Reproduction steps

1. Click the "Deploy to Netlify" button for Hugo Site Starter on the [Start with a Template](https://www.netlifycms.org/docs/start-with-a-template/) page in the docs.
2. Update the configuration for the `date` field on a `post` to store just a date.

```diff
-       - {label: "Publish Date", name: "date", widget: "datetime"}
+       - {label: "Publish Date", name: "date", widget: "datetime", format: "YYYY-MM-DD", timeFormat: false}
```

3. Create and save a new post through the CMS with a Publish Date of 2020-04-20. Note that this will be persisted correctly as `2020-04-20` in the generated markdown.
4. Refresh the page, see that the CMS UI has loaded the Publish Date as 2020-04-19.

**Comment by MohamedBechir** [comment]
@erezrokah  can i work on it ?

**Comment by nholden** [comment]
I started digging into this a little bit. I think if we want the `<DateTime>` component to automatically use UTC for date-only formats, we'd need some way to identify when `format` doesn't contain any time elements. I'm not sure the best way to do that, given that there are so many ways to define formats with [moment.js tokens](https://momentjs.com/docs/#/displaying/format/).

I'm curious what sort of appetite we'd have for to allow the `datetime` widget to accept an optional `pickerUtc` option. It would default to `false`, but when it's `true`, we'd instantiate the `<DateTime>` component with `utc: true`. I don't *love* the idea of adding a new option, but it seems more explicit and resilient than trying to do some magic with the `format` option.

What do you think?

**Comment by erezrokah** [comment]
This will require some digging into, comparing the implementation of the two widgets and making sure we don't break existing behaviour.
With that said @MohamedBechir thank you for suggesting. I've assigned you to the issue, please share your thoughts/findings here as @nholden had a good suggestion.
I think this boils down to "what should we do when the persisted value doesn't contain timezone information?"

**Comment by nholden** [comment]
> I think this boils down to "what should we do when the persisted value doesn't contain timezone information?"

In the context of the `datetime` widget UI, I think what we may be concerned with is "what should we do when the persisted value doesn't contain time."

From my understanding, the `<DateTime>` component used in the `datetime` widget UI will only use your local timezone or UTC (when we pass in the `utc: true` option). It will translate times persisted in other timezones to your local timezone or UTC, but it won't display the persisted timezone.

**Comment by nholden** [comment]
👋Hi, @MohamedBechir! Have you started digging into this one yet? I'm happy to try opening a PR sometime this week if you don't have the chance.

**Comment by erezrokah** [comment]
Hi @nholden, I've assigned you to that issue, thanks! @MohamedBechir we have plenty of other possible contributions to be made, so let me know if you have something in mind.

**Comment by guillett** [comment]
I have been investigating another datetime related issue but I thought it was worth sharing it with you.

De-serialization is delegated to https://www.npmjs.com/package/yaml. It generates a Date in memory.

I [tried to update the in-memory value in the widget](https://github.com/netlify/netlify-cms/blob/master/packages/netlify-cms-widget-datetime/src/DateTimeControl.js#L57-L60) but it set the entry as dirty (it triggers a DRAFT_CHANGE_FIELD action).


My context:

I have
```yaml
name: Cool product
start: 2019-03-07
```
with a 
```yaml
fields:
- label: Nom
  name: title
  widget: string
- label: Start
  name: start
  widget: datetime
  format: YYYY-MM-DD
  dateFormat: YYYY-MM-DD
  timeFormat: false
  default: ''
```

When I update the string the date get persisted as `2019-03-07T00:00:00.000Z` because that value is never touched.

Maybe [`registerWidgetValueSerializer`](https://github.com/netlify/netlify-cms/blob/master/packages/netlify-cms-core/index.d.ts#L219) should be used to ensure that dates also get touched and appropriately serialized.

**Comment by equivalentideas** [comment]
Thanks for all the work on this, we're experiencing this bug and it's setting some important dates off unexpectedly for authors.

Do you have a workaround that's working for you @nholden ?

**Comment by raniesantos** [comment]
Hi, is this the right issue to follow if I'm trying to shorten the dates the CMS outputs to just year, month and date? (`date: 2018-08-31`)

**Comment by guillett** [comment]
@raniesantos I would recommend you #3702. I put a hack to get the output you want.

**Comment by nholden** [comment]
Thanks for sharing, @guillett! As I've dug into the `datetime` widget and the underlying `<DateTime>` picker component, I've been reminded that time is really complicated. 😅

I think it was a good call to open up the separate https://github.com/netlify/netlify-cms/issues/3702 issue for what you described. If you have any specific questions as you dig in, feel free to @ me, and I'd be happy to take a look and share anything relevant that I've learned.

In the meantime, I just opened https://github.com/netlify/netlify-cms/pull/3721 to address the specific issue I described in the original description.
